crates/oxc_linter/src/lib.rs

```diff
@@ -49,16 +56,23 @@ impl Linter {
     }
 
     #[must_use]
-    pub fn run(&self, semantic: &Rc<Semantic>) -> Vec<Error> {
-        let ctx = LintContext::new(semantic.clone());
+    pub fn run(&self, semantic: &Rc<Semantic>, source_text: &str) -> LintRunResult {
+        let mut ctx = LintContext::new(semantic.clone());
 
         for node in semantic.nodes().iter() {
             for rule in &self.rules {
-                rule.run(node, &ctx);
+                if let Some(fix) = rule.run(node, &ctx) {
+                    ctx.push_fix(fix);
+                }
             }
         }
 
-        ctx.into_diagnostics()
+        let mut fixed_source = String::from(source_text);
+        for fix in ctx.fixes() {
+            fixed_source = fix.apply(&fixed_source);
+        }
```

Comment:
This is very naive and obviously would not cover a lot of cases, particularly around conflicting rules. We can probably track the last fix `Span` here and add the additional check for overlapping rules

Reply:
Yes, let's setup proper testing infrastructure before implementing all the edge cases for code fixes.

Reply:
Would you mind helping with that (or someone else)? Just to get the ground work in, I'm not sure the best way to approach it

Reply:
I'll take at look at the test infra this weekend.